Mathlib/Topology/ContinuousFunction/FunctionalCalculus.lean

```diff
@@ -0,0 +1,696 @@
+
++ for `R := ℂ`, we choose `p := IsStarNormal`,
++ for `R := ℝ`, we choose `p := IsSelfAdjoint`,
++ for `R := ℝ≥0`, we choose `p := (0 ≤ ·)`.
+-/
+class ContinuousFunctionalCalculus (R : Type*) {A : Type*} (p : outParam (A → Prop))
+    [CommSemiring R] [StarRing R] [MetricSpace R] [TopologicalSemiring R] [ContinuousStar R]
+    [Ring A] [StarRing A] [TopologicalSpace A] [Algebra R A] where
+  /-- The star algebra homomorphisms of the continuous functional calculus. -/
+  toStarAlgHom {a} (ha : p a) : C(spectrum R a, R) →⋆ₐ[R] A
+  /-- The continuous functional calculus is a closed embedding. -/
+  hom_closedEmbedding {a} (ha : p a) : ClosedEmbedding <| toStarAlgHom ha
+  /-- The continuous functional calculus extends the polynomial functional calculus. -/
+  hom_id {a} (ha : p a) : toStarAlgHom ha ((ContinuousMap.id R).restrict <| spectrum R a) = a
+  /-- The continuous functional calculus satisfies the spectral mapping property. -/
+  hom_map_spectrum {a} (ha : p a) : ∀ f, spectrum R (toStarAlgHom ha f) = Set.range f
+  /-- The continuous functional calculus preserves the predicate. -/
+  predicate_hom {a} (ha : p a) : ∀ f, p (toStarAlgHom ha f)
+
+/-- A class guaranteeing that the continuous functional calculus is uniquely determined by the
+properties that it is a continuous star algebra homomorphism mapping the (restriction of) the
+identity to `a`. This is the necessary tool used to establish `cfcSpec_comp` and the more common
+variant `cfc_comp`.
+
+This class will have instances in each of the common cases `ℂ`, `ℝ` and `ℝ≥0` as a consequence of
+the Stone-Weierstrass theorem. -/
+class UniqueContinuousFunctionalCalculus (R A : Type*) [CommSemiring R] [StarRing R]
+    [MetricSpace R] [TopologicalSemiring R] [ContinuousStar R] [Ring A] [StarRing A]
+    [TopologicalSpace A] [Algebra R A] : Prop where
+  eq_of_continuous_of_map_id (a : A) (φ ψ : C(spectrum R a, R) →⋆ₐ[R] A)
+    (hφ : Continuous φ) (hψ : Continuous ψ)
+    (h : φ (.restrict (spectrum R a) <| .id R) = ψ (.restrict (spectrum R a) <| .id R)) :
+    φ = ψ
+
+variable {R A : Type*} {p : A → Prop} [CommSemiring R] [StarRing R] [MetricSpace R]
+variable [TopologicalSemiring R] [ContinuousStar R] [TopologicalSpace A] [Ring A] [StarRing A]
+variable [Algebra R A] [ContinuousFunctionalCalculus R p]
+
+lemma StarAlgHom.ext_continuousMap [UniqueContinuousFunctionalCalculus R A]
+    (a : A) (φ ψ : C(spectrum R a, R) →⋆ₐ[R] A) (hφ : Continuous φ) (hψ : Continuous ψ)
+    (h : φ (.restrict (spectrum R a) <| .id R) = ψ (.restrict (spectrum R a) <| .id R)) :
+    φ = ψ :=
+  UniqueContinuousFunctionalCalculus.eq_of_continuous_of_map_id a φ ψ hφ hψ h
+
+section cfcSpec
+
+variable {a : A} (ha : p a)
+
+-- Note: since `spectrum R a` is closed, we may always extend `f : C(spectrum R a, R)` to a function
+-- of type `C(R, R)` by the Tietze extension theorem (assuming `R` is either `ℝ`, `ℂ` or `ℝ≥0`).
```

Comment:
Is this note still relevant?

Reply:
I think it's nice to leave it as a reminder to the reader, especially since we now have the nice `TietzeExtension` class, but it's certainly less relevant than it used to be. I can remove it if you prefer.

Reply:
No problem with keeping it, just wanted to make sure it was supposed to be here.